src/components/search/search.jsx

```diff
@@ -0,0 +1,367 @@
+import {
+	useState,
+	forwardRef,
+	useRef,
+	createContext,
+	useContext,
+	Children,
+} from 'react';
+import { cn } from '@/utilities/functions';
+import { Search } from 'lucide-react';
+import Loader from '../loader';
+import {
+	textSizeClassNames,
+	variantClassNames,
+	disabledClassNames,
+	sizeClassNames,
+	IconClasses,
+} from './styles';
+import {
+	autoUpdate,
+	flip,
+	FloatingPortal,
+	offset,
+	size as floatingSize,
+	useDismiss,
+	useFloating,
+	useInteractions,
+} from '@floating-ui/react';
+
+const SearchContext = createContext();
+
+const useSearchContext = () => {
+	return useContext( SearchContext );
+};
+
+const SearchBox = forwardRef(
+	(
+		{
+			className, // Custom class names for additional styling
+			size = 'sm', // Size of the SearchBox (default is 'sm')
+			open = false, // Controls whether the dropdown is open or not (default is false)
+			onOpenChange = () => {}, // Callback function to handle the open state change
+			...props // Any additional props that are passed
+		},
+		ref
+	) => {
+		const { refs, floatingStyles, context } = useFloating( {
+			open,
+			onOpenChange,
+			placement: 'bottom-start',
+			whileElementsMounted: autoUpdate,
+			middleware: [
+				offset( size === 'sm' ? 4 : 6 ),
+				flip( { padding: 10 } ),
+				floatingSize( {
+					apply( { rects, elements, availableHeight } ) {
+						elements.floating.style.maxHeight = `${ availableHeight }px`;
+						elements.floating.style.width = `${ rects.reference.width }px`;
+						elements.floating.style.fontFamily =
+							window.getComputedStyle(
+								elements.reference
+							).fontFamily; // Retain parent's font family
+					},
+				} ),
+			],
+		} );
+		const dismiss = useDismiss( context );
+
+		const { getReferenceProps, getFloatingProps } = useInteractions( [
+			dismiss,
+		] );
+
+		return (
+			<SearchContext.Provider
+				value={ {
+					size, // Pass size to context
+					open, // Pass open state to context
+					onOpenChange, // Pass onOpenChange handler to context
+					refs, // Pass refs to context
+					floatingStyles, // Pass floating styles to context
+					context, // Pass context to context (nested contexts)
+					getReferenceProps, // Reference props for floating elements
+					getFloatingProps, // Floating props for dropdown content
```

Comment:
Maybe you shouldn't leave comments here, since you explain the props in the readme file.